src/coreclr/debug/dbgutil/machoreader.cpp

```diff
@@ -413,23 +392,27 @@ MachOReader::EnumerateModules(mach_vm_address_t address, mach_header_64* header)
     }
     for (int i = 0; i < dyldInfo.infoArrayCount; i++)
     {
-        mach_vm_address_t imageAddress = (mach_vm_address_t)imageInfos[i].imageLoadAddress;
-        const char* imageFilePathAddress = imageInfos[i].imageFilePath;
+        // Ignore any errors and continue to next module
+        CreateModule(imageInfos[i].imageLoadAddress, imageInfos[i].imageFilePath);
+    }
+    return true;
+}
 
-        std::string imagePath;
-        if (!ReadString(imageFilePathAddress, imagePath))
-        {
-            Trace("ERROR: Failed to read image name at %p\n", imageFilePathAddress);
-            continue;
-        }
-        MachOModule module(*this, imageAddress, nullptr, &imagePath);
-        if (!module.ReadHeader())
-        {
-            continue;
-        }
-        Trace("MOD: %016llx %08x %s\n", imageAddress, module.Header().flags, imagePath.c_str());
-        VisitModule(module);
+bool
+MachOReader::CreateModule(const struct mach_header*	imageAddress, const char* imageFilePathAddress)
```

Comment:
nit: `TryRegisterModule` may be a better name since this never returns the module - rather visitmodule registers it

Reply:
I'll catch this next time. TryRegisterModule is a better name.